controllers/iperf3/controller.go

```diff
@@ -42,46 +39,63 @@ type Reconciler struct {
 // Reconcile Iperf3 Job Requests
 func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
-	log := r.Log.WithValues("iperf3", req.NamespacedName)
 
 	var cr perfv1alpha1.Iperf3
 	if err := r.K8S.Client.Get(ctx, req.NamespacedName, &cr); err != nil {
-		//log.Error(err, "Unable to fetch Iperf3 CR")
 		return ctrl.Result{}, k8s.IgnoreNotFound(err)
 	}
 
-	crRef, err := reference.GetReference(r.K8S.Scheme, &cr)
-	if err != nil {
-		log.Error(err, "Unable to get reference to Iperf3 CR")
+	// Run to one completion
+	if cr.Status.Completed {
+		return ctrl.Result{}, nil
+	}
+
+	cr.Status.Running = true
+	if err := r.K8S.Client.Status().Update(ctx, &cr); err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if err := r.newServerDeployment(ctx, &cr, crRef); err != nil {
-		r.K8S.EventRecorder.Eventf(crRef, corev1.EventTypeWarning, k8s.CreateFailed,
-			"Unable to create iperf3 Server Deployment: %v", err)
+	if err := r.K8S.CreateWithReference(ctx, newServerDeployment(&cr), &cr); err != nil {
 		return ctrl.Result{}, err
 	}
 
-	if err := r.newServerService(ctx, &cr, crRef); err != nil {
-		r.K8S.EventRecorder.Eventf(crRef, corev1.EventTypeWarning, k8s.CreateFailed,
-			"Unable to create iperf3 Server Service: %v", err)
+	if err := r.K8S.CreateWithReference(ctx, newServerService(&cr), &cr); err != nil {
 		return ctrl.Result{}, err
 	}
 
 	serverReady, err := r.serverDeploymentReady(&cr)
 	if err != nil {
-		r.K8S.EventRecorder.Eventf(crRef, corev1.EventTypeWarning, k8s.CreateFailed,
-			"Unable to determine iperf3 Server Deployment state: %v", err)
 		return ctrl.Result{}, err
 	}
 	if !serverReady {
-		// Wait for the client pod to the deployment to be ready
+		// Wait for deployment to be ready
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	if err := r.newClientPod(ctx, &cr, crRef); err != nil {
-		r.K8S.EventRecorder.Eventf(crRef, corev1.EventTypeWarning, k8s.CreateFailed,
-			"Unable to create iperf3 client pod: %v", err)
+	if err := r.K8S.CreateWithReference(ctx, newClientPod(&cr), &cr); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	clientCompleted, err := r.clientPodFinished(&cr)
```

Comment:
Please resemble the method name in the variable: 
`clientPodFinished, err := r.clientPodFinished(&cr)`

Reply:
done